collectors/framework/framework.go

```diff
@@ -256,6 +256,23 @@ func (a *AvroDatum) transform(nodeInfo collectors.NodeInfo) (producers.MetricsMe
 		return pmm, err
 	}
 
+	// Convert message-level labels to datapoint-level tags
+	labels := pmm.Dimensions.Labels
+	if labels != nil {
```

Comment:
You should not need to check for `nil` here.  Using `range` on a `nil` map should be ok.  This could just be

```
for n, d := range pmm.Dimensions.Labels {
```